File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/instances/DisplayInstance.java
package net.kdt.pojavlaunch.instances;

import java.io.File;

public class DisplayInstance {
    public String name;
    public String versionId;
    public String icon;
    protected transient File mInstanceRoot;

    protected DisplayInstance() {
    }

    protected void sanitize() {
        sanitizeIcon();
    }

    protected File getInstanceIconLocation() {
        return new File(mInstanceRoot, "icon.webp");
    }

    private void sanitizeIcon() {
        if (!InstanceIconProvider.hasStaticIcon(icon)) {
            icon = InstanceIconProvider.FALLBACK_ICON_NAME;
        }
    }
}

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/instances/DisplayInstance.kt
package net.kdt.pojavlaunch.instances

import java.io.File

open class DisplayInstance {
    @Transient
    @JvmField var mInstanceRoot: File? = null
    @JvmField var name: String? = null
    @JvmField var versionId: String? = null
    @JvmField var icon: String? = null

    protected open fun sanitize() {
        sanitizeIcon()
    }

    protected constructor()

    protected fun getInstanceIconLocation(): File {
        return File(mInstanceRoot, "icon.webp")
    }

    private fun sanitizeIcon() {
        if (!InstanceIconProvider.hasStaticIcon(icon)) {
            icon = InstanceIconProvider.FALLBACK_ICON_NAME
        }
    }
}
